### PR TITLE
Add SCSS support

### LINKED ISSUE #59: Support SCSS
Now supported through a custom parser in [autoprefixer@6.0.0](https://github.com/postcss/autoprefixer/releases/tag/6.0.0):
https://github.com/postcss/postcss/#custom-syntaxes
https://github.com/postcss/postcss-scss

I don't have time to do this right now, but PR welcome :)

### PR DESCRIPTION
**Pull request by gjjones**
Fixes #59

### CODE REVIEW

File: Autoprefixer.py [hunk 1]
```diff
@@ -44,9 +44,11 @@ def run(self, edit):
 
 	def prefix(self, data):
 		try:
+			isCSS = splitext(basename(self.view.settings().get('syntax')))[0] == 'CSS'
```
Comment (sindresorhus) [inline Autoprefixer.py]: Just put this inline on line 51.

Comment (sindresorhus) [inline Autoprefixer.py]: `splitext(basename(self.view.settings().get('syntax')))[0]` is used twice and could be put in a separate function that accepts a view and returns the syntax name.

File: autoprefixer.js [hunk 2]
```diff
@@ -3,12 +3,17 @@ var getStdin = require('get-stdin');
...
 var autoprefixer = require('autoprefixer');
 var postcssSafeParser = require('postcss-safe-parser');
+var postcssScssParser = require('postcss-scss');
 
 getStdin().then(function (data) {
 	var opts = JSON.parse(process.argv[2]);
 
+	var postcssParser = opts.is_css ?
+							postcssSafeParser :
+							postcssScssParser;
```
Comment (sindresorhus) [inline autoprefixer.js]: Should be on one line.

File: Autoprefixer.py [hunk 3]
```diff
@@ -21,8 +21,8 @@ def get_setting(view, key):
 		settings = sublime.load_settings('Autoprefixer.sublime-settings')
 	return settings.get(key)
 
-def is_css(view):
-	return splitext(basename(view.settings().get('syntax')))[0] == 'CSS'
+def is_style_syntax(view):
+	return splitext(basename(view.settings().get('syntax')))[0] in ['CSS', 'Sass', 'SCSS']
```
Comment (sindresorhus) [inline Autoprefixer.py]: I don't think the parser supports `Sass`, just `SCSS`.

File: readme.md [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+> Sublime plugin to prefix your CSS or SCSS
 
 You shouldn't have to care about vendor prefixes. Now you don't have too.
 
 ![](screenshot.gif)
 
 Adding prefixes manually is a chore. It's also hard to keep track of where and which prefixes are needed. This plugin uses the [Autoprefixer](https://github.com/ai/autoprefixer) library to prefix properties and values according to the [Can I Use](http://caniuse.com/) database. Which means it will only add the necessary prefixes and not bloat your stylesheet. It even lets you specify what browsers you want to target. In addition it will remove existing prefixes which are no longer needed.
 
-Works only with CSS and not preprocessors like Sass or LESS.
+Works with CSS and SCSS, and does not currently support other preprocessors.
```
Comment (sindresorhus) [inline readme.md]: > Works with CSS and SCSS, but not other preprocessors.

File: Autoprefixer.py (hunk 3 above)
Comment (gjjones) [inline Autoprefixer.py]: These refer to syntax packages, and the Sass package also highlights the SCSS syntax.

I'd rather we check the file extension instead. Neither of these syntaxes ships by default, so it doesn't make sense to be aware of them in here.

Comment (sindresorhus) [inline Autoprefixer.py]: > I'd rather we check the file extension instead.

👍

**Comment by hudochenkov** [comment]
Plugin will stop work after this PR on unsaved files. If user just created new view, set syntax to CSS, then plugin will not work on that view until user will save file.

**Comment by gjjones** [comment]
Would it be preferred to detect syntax OR file extension for the syntaxes that ship by default, and just detect file extension for SCSS then? It seems like voodoo to the user to detect the syntax in one case but not the other. Could always add a setting to allow users to add which SCSS syntax they're using, then check against that in the case of unsaved SCSS files in addition to the file extension support.

**Comment by gjjones** [comment]
@hudochenkov, did you have opinions on [this](https://github.com/hudochenkov/sublime-postcss-sorting/blob/master/sorting.js#L10) being a better alternative to how this PR adds the SCSS parser? It would save from needing to pass along which syntax is used, but it could potentially break for people relying on the "fixes your code" aspect of postcss-safe-parser.

**Comment by hudochenkov** [comment]
`postcss-scss` can parse both SCSS and vanilla CSS. BTW, neither of PostCSS and Autoprefixer uses `postcss-safe-parser` by default.

If you want to use `postcss-scss` and `postcss-safe-parser` I suggest to do something like: if view syntax contains `SCSS` in its name, then use `postcss-scss`, otherwise use `postcss-safe-parser`.

**Comment by gjjones** [comment]
@sindresorhus, given that the splintered use of [Sass](https://packagecontrol.io/packages/Sass) vs [SCSS](https://packagecontrol.io/packages/SCSS) for syntax highlighting, I don't see the point of assuming syntax without allowing the user to provide it as input. Did you have a preference towards keeping `postcss-safe-parser`? Can add the settings input for syntax if so, otherwise it seems like this package would be better off switching to just using `postcss-scss`.

**Comment by hudochenkov** [comment]
There is one more popular Sass/SCSS syntax highlighting package https://packagecontrol.io/packages/Syntax%20Highlighting%20for%20Sass

**Comment by gjjones** [comment]
[Here](https://github.com/gjjones/sublime-stylefmt/commit/984eaa07b4503e1667a2f5b3d66656dcd5f50d5c) is an example of using a setting to specify the syntax, and default to `SCSS`. Seems the smallest impact since it still assumes `SCSS`, while also allowing an override.

**Comment by sindresorhus** [comment]
Not interested in options. This package should just work, even on unsaved buffers. `postcss-safe-parser` is used as it's more lenient and will work even with some invalid CSS, which often happens you write CSS. Just check both extension and syntax.

**Comment by sindresorhus** [comment]
Looks great! Thank you @gjjones :)

tap-dancing

**Pull request merged**